base/logger/R/print2string.R

```diff
@@ -0,0 +1,22 @@
+#' Convert the printed output of an object to a string. This is
+#' particularly useful for passing complex objects (e.g. matrices,
+#' data.frames) to simple printing functions like
+#' [PEcAn.logger::logger.info()] or [base::message()].
+#'
+#' Note that for this to work properly in the `PEcAn.logger`
+#' functions, you should always add the `wrap = FALSE` argument, and
+#' probably add a newline (`"\n"`) before the output of this function.
+#'
+#' @param x Object to print
+#' @param ... Additional arguments to [base::print()].
+#' @return Output of `print(x, ...)`, captured as string
+#' @author Alexey Shiklomanov
+#' @examples
+#' logger.info("First few rows of Iris:\n", print2string(iris[1:10, -5]), wrap = FALSE)
+#' df <- data.frame(test = c("download", "process", "plot"), status = c(TRUE, TRUE, FALSE))
+#' logger.debug("Current status:\n", print2string(df, row.names = FALSE), wrap = FALSE)
+#' @export
+print2string <- function(x, ...) {
+  cout <- capture.output(print(x, ...))
+  paste(cout, collapse = "\n")
+}
```

Comment:
Here's what this looks like in action:

```r
PEcAn.logger::logger.info(
  "First few lines of iris:\n",
  PEcAn.logger::print2string(iris[1:5, ]),
  wrap = FALSE
)
# 2018-11-19 16:37:44 INFO   [console] : First few lines of iris:
#    Sepal.Length Sepal.Width Petal.Length Petal.Width Species
# 1          5.1         3.5          1.4         0.2  setosa
# 2          4.9         3.0          1.4         0.2  setosa
# 3          4.7         3.2          1.3         0.2  setosa
# 4          4.6         3.1          1.5         0.2  setosa
# 5          5.0         3.6          1.4         0.2  setosa 
```

You can also pass arguments to `print`:

```r
PEcAn.logger::logger.info(
  "First few lines of iris:\n",
  PEcAn.logger::print2string(iris[1:5, ], row.names = FALSE),
  wrap = FALSE
)
# 2018-11-19 16:38:59 INFO   [console] : First few lines of iris:
#   Sepal.Length Sepal.Width Petal.Length Petal.Width Species
#           5.1         3.5          1.4         0.2  setosa
#           4.9         3.0          1.4         0.2  setosa
#           4.7         3.2          1.3         0.2  setosa
#           4.6         3.1          1.5         0.2  setosa
#           5.0         3.6          1.4         0.2  setosa 
```